### DISCUSSION TITLE
How to start using it in vim?

### DISCUSSION THREAD

**Discussion by GitMensch** [OCamlPro/superbol-studio-oss#345] [Q&A]
We do have https://github.com/OCamlPro/superbol-studio-oss/wiki/SuperBOL-LSP but apart from the fact that it would be nice to have superbol-free binaries (you have them packaged in the vsix after all), so a local build is not needed: how to get started when neither coc nor vim-lsp are used (= think about a user that only knows vim in general).

And... shouldn't there be a configuration of copybook directories and extensions in the config?

So after this is setup: how to use it? (a link to a good explanation would be fine, too)

**Comment by emilienlemaire** [comment]
Hello Simon,
Thank you for your questions :)

I will answer regarding the vim part only. Sadly you cannot use any lsp of any kind in vim natively without any plugin. So I wrote the tutorial using the most popular lsp client plugins for vim. I could add links to the plugins installation instructions page, but not much more. If you wish to have native support of the lsp within vim, using neovim is the only alternative.

To configure the copybook directories and extensions you should use the `superbol.toml` file, usually you should put it at the root of your project. You can generate a default one with `superbol project init` and then edit it to your liking.

Once you have used the given configurations the lsp should start itself whenever you open a COBOL file.

**Comment by GitMensch** [comment]
How to update the LSP using opam?
Can we use the native binaries from the vsix as well?

... and how to add it to https://opencode.ai/docs/lsp/#custom-lsp-servers ?

**Reply by emilienlemaire** [reply]
Hello @GitMensch,
Assuming you installed it according to https://github.com/OCamlPro/superbol-studio-oss/wiki/SuperBOL-LSP , you should be able to checkout the version you'd like to install from git, and then do `drom install --switch superbol-studio -y`, it will update it to the version you checked out.

I don't really know if the native binaries can be extracted from the `vsix`, but they should be the same as the one built and installed with `opam`.

As for `opencode`, I have never heard about it before today, but looking into the page you sent, I would assume that the configuration should look something like that:
```json
{
  "$schema": "https://opencode.ai/config.json",
  "lsp": {
    "custom-lsp": {
      "command": [
        "opam",
        "exec",
        "--switch",
        "superbol-studio",
        "--",
        "superbol-free",
        "lsp",
        "--force-syntax-diagnostics"
      ],
      "extensions": [".custom"]
    }
  }
}
```
